### PR TITLE
[rfr]Fix orchestration output

### PR DESCRIPTION
**Pull request by pratikmallya**

### CODE REVIEW

File: commands/orchestrationcommands/stackcommands/adopt_test.go [hunk 1]
```diff
@@ -36,3 +40,57 @@ func TestAdoptServiceClientType(t *testing.T) {
...
+		Ctx: &handler.Context{
+			ServiceClient: client.ServiceClient(),
+		},
+	}
+
+	resource := &handler.Resource{
+		Params: &paramsAdopt{},
+	}
+
+	expected := "{\"ID\":\"16ef0584-4458-41eb-87c8-0dc8d5f66c87\",\"Links\":[{\"Href\":\"http://168.28.170.117:8004/v1/98606384f58drad0bhdb7d02779549ac/stacks/stackcreated/16ef0584-4458-41eb-87c8-0dc8d5f66c87\",\"Rel\":\"self\"}]}"
```
Comment (jrperritt) [inline commands/orchestrationcommands/stackcommands/adopt_test.go]: The PR looks great. There's just one thing confusing me: I see above you're flattening the `Links` field in the `adopt` `PreCSV`, and below you're calling that method, so why are there arrays in the JSONified `PreCSV` output above?

Comment (pratikmallya) [inline commands/orchestrationcommands/stackcommands/adopt_test.go]: Ah, good question. What is happening is that the `FlattenMap` method does not handle the keys correctly if they're `gophercloud.Link` type. In the code, the method `stackSingle` (or something similar) is called on the result to convert the `Links` to a `map[string]interface{}` or similar type.

Comment (jrperritt) [inline commands/orchestrationcommands/stackcommands/adopt_test.go]: Excellent :+1:

**Comment by jrperritt** [comment]
+2

**Pull request merged**